### ISSUE TITLE
support linked documents

### ISSUE THREAD

**Issue by f3rdy** [justmiles/go-markdown2confluence#19]
Hi, thanks for the tool. At a first glance it looks promising and helpful.

I've got a structure like this in my project:

```
project_root/readme.md
project_root/documentation/do_this.md
project_root/documentation/do_that.md
```

I use the readme.md moreover to link to the subdocuments and use it primarily as an index or TOC file. From the subdocuments, I link back to the index or to other subdocuments. 

First problem I encounter is, that the the links between the project root directory and subdirectories are not translated into confluence pages/subpages and therefore do not work.

I think, I could flatten the structure to put the readme.md into the documentation subdirectory of the source code and just use the project root readme.md to link to the documentation/readme.md. That might do it.

A different case would be to have links to (other) source code files in my markdown. It would be cool to have a possibility to change relative links to local files at least by absolute links to the current repository. Sure, I can do that directly in the markdown, but wouldn't it be better to have this in the tooling?

Thanks,
ferdy

**Comment by justmiles** [comment]
Hey @f3rdy. I like the idea and agree it would certainly be useful, but I'm not really sure how to implement it. The API doesn't have an ID for the confluence document until after it is created. You could probably look at each document and map dependencies so that each document knows the actual confluence link it needs to point to before rendering - but what if you have two documents that link to one-another. Then again, we could map all the dependencies, upload, then edit and re-upload anything linking.

Initial thoughts are that it's a sizable chunk of work for the value. Definitely happy to merge pull requests that solve for it but this doesn't seem like something I can tackle right now. I'll leave this open in the hopes that someone else has the motivation to take it on. Thanks for opening the ticket!

**Comment by Skeeve** [comment]
Are Document IDs really required?

When I created 2 testpages, the link generated in the page looks like this:

```html
<ac:link><ri:page ri:content-title="TestPage2" /></ac:link>
```

**Comment by mschmieder** [comment]
Please have a look at the #40. I did take a shot at this...

**Comment by justmiles** [comment]
This is available in [v4.0.0-alpha-2](https://github.com/justmiles/go-markdown2confluence/releases/tag/v4.0.0-alpha-2)
